packages/core/src/NavigationContainer.tsx

```diff
@@ -92,10 +93,10 @@ const Container = React.forwardRef(function NavigationContainer(
     getPartialState(initialState)
   );
 
+  const { listeners, addNavigationListener } = useFocusedListeners();
+
   const dispatch = (action: NavigationAction) =>
-    context.performTransaction(() => {
-      actionListeners[0](action, undefined, null);
-    });
+    listeners[0](navigation => navigation.dispatch(action));
```

Comment:
Still think there's no reason not to loop over the listeners.

Reply:
Should wrap this in curly braces so the return value of `dispatch` isn't a boolean

Reply:
I think since we're absolutely sure that here cannot be more than one listener it's ok